### PR TITLE
Test case Automation: CEPH-83594299 - Cloning and Flattening a …

### PR DESCRIPTION
**Pull request by chaitanyadommeti**
…Group Snapshot

# Description
Code for test [CEPH-83594299 - Cloning and Flattening a Group Snapshot](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83594299)

Modified the suite: suites/squid/rbd/tier-2_rbd_regression.yaml

Pass log: 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VO6ZMD/

### CODE REVIEW

**Comment by chaitanyadommeti** [comment]
> Please add success logs.
Success log after review comments implementation:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VO6ZMD/

**Comment by chaitanyadommeti** [comment]
> Please add success logs.
Success log after implementation of review comments
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VO6ZMD/

**Comment by chaitanyadommeti** [comment]
> Please add success logs.
Success logs after implementation of review comments:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VO6ZMD/

File: suites/squid/rbd/tier-2_rbd_regression.yaml [hunk 1]
```diff
@@ -238,3 +238,20 @@ tests:
       module: test_rbd_python_module.py
       polarion-id: CEPH-83574791
       name: Test image creation, write and read data using python rbd module
+
+  - test:
```
Comment (sunilangadi2) [inline suites/squid/rbd/tier-2_rbd_regression.yaml:242]: please add this test in this suite as it's related to the group feature
https://github.com/red-hat-storage/cephci/blob/master/suites/squid/rbd/tier-2_rbd_group.yaml

File: tests/rbd/test_rbd_clone_flatten_group_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+                return 1
+            else:
+                log.info("Flattened clone removed successfully")
+
+    return 0
+
+
+def run(**kw):
+    """Test to verify Cloning and Flattening a Group Snapshot
+    This test verifies clonign and flattening a group snapshot
```
[the remark this answers is not in the document]
Comment (sunilangadi2) [reply tests/rbd/test_rbd_clone_flatten_group_snapshot.py:252]: spelling mistake cloning

File: tests/rbd/test_rbd_clone_flatten_group_snapshot.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+            client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+        else:
+            client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+        rbd_obj = initial_rbd_config(**kw)
+        pool_types = rbd_obj.get("pool_types")
+        if rbd_obj:
+            log.info("Executing test on Replication and EC pool")
+            if clone_flatten_group_snapshot(rbd_obj, client, **kw):
+                return 1
+
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_clone_flatten_group_snapshot.py:272]: add `log.info` for the successful message.

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: tests/rbd/test_rbd_clone_flatten_group_snapshot.py (hunk 3 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_clone_flatten_group_snapshot.py:279]: Added

File: suites/squid/rbd/tier-2_rbd_regression.yaml (hunk 1 above)
Comment (chaitanyadommeti) [reply suites/squid/rbd/tier-2_rbd_regression.yaml]: Addressed

File: tests/rbd/test_rbd_clone_flatten_group_snapshot.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+            # Verify the cloned image’s parent information after flattening using rbd info
+            out = rbd.info(**info_kw)
+            out_json = json.loads(out[0])
+
+            if "parent" in out_json:
+                log.error("parent information still exist for flattened clone")
+                return 1
+            else:
+                log.error(
```
[the remark this answers is not in the document]
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_clone_flatten_group_snapshot.py]: Handled in new commit and retested. success log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8B51O2/

File: tests/rbd/test_rbd_clone_flatten_group_snapshot.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_clone_flatten_group_snapshot.py]: Addressed

**Review by sunilangadi2 — APPROVED** [review]
changes LGTM.

File: tests/rbd/test_rbd_clone_flatten_group_snapshot.py [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    except Exception as e:
+        log.error(f"Test Cloning and Flattening a Group Snapshot failed: {str(e)}")
+        return 1
+
+    finally:
+        cluster_name = kw.get("ceph_cluster", {}).name
+        if "rbd_obj" not in locals():
+            rbd_obj = Rbd(client)
+        obj = {cluster_name: rbd_obj}
+        cleanup(pool_types=pool_types, multi_cluster_obj=obj, **kw)
```
Comment (aarsharm) [inline tests/rbd/test_rbd_clone_flatten_group_snapshot.py:289]: Here if pool_types is not defined (e.g., due to a failure in initial_rbd_config()), the cleanup call will throw an UnboundLocalError.
Suggestion: Initialize pool_types to an empty list or include a check before cleanup:

**Review by aarsharm — APPROVED** [review]

**Review by sunilangadi2 — APPROVED** [review]
LGTM.

**Review by sunilkumarn417 — APPROVED** [review]
Approving the PR with already provided ACKs.

**Pull request merged**